### PR TITLE
Add solana-install design proposal

### PR DESCRIPTION
**Pull request by mvines**
Design proposal for #3222

### CODE REVIEW

**Comment by mvines** [comment]
💔 can't use clap's yaml facility due to a stale rust-yml dependency that triggers `cargo audit`

File: install/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("update")
+                .about("checks for an update, and if available downloads and applies it")
+                .setting(AppSettings::DisableVersion),
+        )
+        .subcommand(
+            SubCommand::with_name("run")
+                .about("Runs a program while periodically checking and applying software updates")
```
Comment (garious) [inline install/src/main.rs]: A slippery slope?  What's the scope of `solana-install`? An installer/updater or everything we need it to be to validate a deployment?

File: book/src/installer.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  pub target: String,                 // Target triple (TARGET)
+  pub commit: String,                 // git sha1 of this update, must match the commit sha1 in the release tar.bz2
+  pub timestamp_secs: u64,            // When the release was deployed (seconds since UNIX EPOC)
+  pub download_url: String,           // Download URL to the release tar.bz2
+  pub download_signature: Signature,  // Signature of the release tar.bz2 file, verify with the Account public key
+}
+
+/// Userdata of an Update Manifest program Account.
+pub struct SignedUpdateManifest {
+  pub manifest: UpdateInfo,
```
Comment (rob-solana) [inline book/src/installer.md]: UpdateManifest?

Comment (mvines) [reply book/src/installer.md]: Sorry I can’t parse, need a couple more words please

Comment (rob-solana) [reply book/src/installer.md]: should this line read `pub manifest: UpdateManifest` ?

Comment (mvines) [reply book/src/installer.md]: Oh yep thanks

**Pull request merged**